pkg/controller/namespace/namespace_controller.go

```diff
@@ -132,7 +132,9 @@ func (nm *NamespaceController) enqueueNamespace(obj interface{}) {
 		glog.Errorf("Couldn't get key for object %+v: %v", obj, err)
 		return
 	}
-	nm.queue.Add(key)
+	// delay processing namespace deletions long enough for the lifecycle admission
+	// plugins to also observe the deletion and stop new objects from being created
+	nm.queue.AddAfter(key, 5*time.Second)
```

Comment:
> Regarding a HA etcd, is the namespace controller getting all its watch events from the master but can try to list in a different etcd member?

The controller only speaks to the API... it gets watch events from some apiserver via some etcd server, but there's no guarantee the list/deletecollection calls it makes go to the same apiserver, or the same etcd server